src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php

```diff
@@ -4,28 +4,73 @@
 
 use Closure;
 use Illuminate\Contracts\Auth\MustVerifyEmail;
-use Illuminate\Support\Facades\Redirect;
+use Illuminate\Contracts\Routing\ResponseFactory;
+use Illuminate\Contracts\Routing\UrlGenerator;
 
 class EnsureEmailIsVerified
 {
+    /**
+     * The response factory instance.
+     *
+     * @var \Illuminate\Contracts\Routing\ResponseFactory
+     */
+    protected $responseFactory;
+
+    /**
+     * The URL generator instance.
+     *
+     * @var \Illuminate\Contracts\Routing\UrlGenerator
+     */
+    protected $urlGenerator;
+
+    /**
+     * Create a new middleware instance.
+     *
+     * @param  \Illuminate\Contracts\Routing\ResponseFactory  $responseFactory
+     * @param  \Illuminate\Contracts\Routing\UrlGenerator  $urlGenerator
+     * @return void
+     */
+    public function __construct(ResponseFactory $responseFactory, UrlGenerator $urlGenerator)
+    {
+        $this->responseFactory = $responseFactory;
+        $this->urlGenerator = $urlGenerator;
+    }
+
     /**
      * Handle an incoming request.
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Closure  $next
      * @param  string|null  $redirectToRoute
-     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     * @return mixed
```

Comment:
Please keep the original return values, and make sure the implementation returns responses.

Reply:
Are you asking me to just revert this line? I believe all returns are responses or redirects.

Reply:
Yes, then.

Reply:
Line reverted.